packages/compass-components/src/components/radio-box-group.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { RadioBoxGroup as LeafygreenRadioBoxGroup } from '@leafygreen-ui/radio-box-group';
+import { Label } from '@leafygreen-ui/typography';
+
+type BasicRadioGroupProps = React.ComponentProps<
+  typeof LeafygreenRadioBoxGroup
+>;
+
+interface RadioBoxGroupWithoutLabel extends BasicRadioGroupProps {
+  id?: string;
+  label: never;
+}
+
+interface RadioBoxGroupWithLabel extends BasicRadioGroupProps {
+  id: string; // We require an id for the label to reference.
+  label: string | JSX.Element;
+}
+
+type RadioBoxGroupProps = RadioBoxGroupWithoutLabel | RadioBoxGroupWithLabel;
+
+function RadioBoxGroup(props: RadioBoxGroupProps): JSX.Element {
+  if (!props.label || !props.id) {
+    return <LeafygreenRadioBoxGroup {...props} />;
+  }
+
+  const { id, label } = props;
+
+  if (typeof label === 'string') {
+    const labelId = `${id}-label`;
+    return (
+      <>
+        <Label htmlFor={id} id={labelId}>
+          {label}
+        </Label>
+        <LeafygreenRadioBoxGroup aria-labelledby={labelId} {...props} />
+      </>
+    );
+  }
+
+  return (
+    <>
+      {label}
+      <LeafygreenRadioBoxGroup {...props} />
+    </>
+  );
+}
```

Comment:
I don't want to contest that there are other ways to factor this composition, but I think here the `FileInput` example is a good argument to actually centralize and standardize those foundational pieces of UI in one place.

One of the reasons why the FileInput is so property-intensive as a React component is because the code was spread around and was then factored in one place with multiple "design" use cases in mind.
I also strongly believe we shouldn't do that again, but a good way to avoid it in the future is to make these foundational bricks our "way" to solve a design need and use them consistently everywhere. There is nothing "connection-like" about our file inputs or radio buttons, and as such there should not be a need to have them different in different forms. 

That involves also influencing design in order to keep the same patterns everywhere: sure technically the file input looks like a "base class" gone wrong and doesn't follow React principles, but before all of that there is no reason to have multiple types and designs of file inputs.

In my opinion for all the form components, we should aim to do the same, where we won't do that we will create divergences, for example, the "Create Collection" dialog is already using different patterns and different components to do the same.

Reply:
This does sound good to me, but how do we make sure this happens? IIRC we discussed a few times already that we are introducing different patterns for similar things in the scope of the same form (i.e., the "hosts" and "url options" multiple fields or labels being sometimes on top of inputs and sometimes on the side which leafygreen [has some opinions on](https://www.mongodb.design/component/text-input/guidelines/) by the way) but these things are still there and because of that my understanding was that having slight variations based on the use-case is something that was accepted

Reply:
That's a good question, the best attempt we have done is start this: https://docs.google.com/document/d/1ZLz_DJYpwTiS_DiPcQMKoLoz2iNxSsMKRTuXXxHb6ak/edit#heading=h.2s9zdv6a57gn

Is more an excuse to talk about that at this point than anything, maybe we invite also design, I guess in general where diverging doesn't give us a clear "advantage" aiming for reuse and consistency makes sense to me.

Reply:
I have a semi-related question to both about the interface of this component, Is clear that in LG this was meant to be a form component looking at the guidelines, but it doesn't have a `label` property as other components have. Here we are adding one but is different from all the other labels for all the other components (is an element). Should we talk with LG first? What if they won't ever reconcile the interfaces?

In the other places where we need `LabelWithInfo` we are using it as a separate element. Should we do the same here for now, and maybe solve the lack of label in a second moment and in a different way?

Reply:
Updated to having an `InlineInfoLink` component with just the icon button + icon in it, and made it not wrap the label. Is this sort of what you were thinking with the composition? Definitely cleaner this way. There are one or two other places where an info link button is used. They are, however, both places where the bootstrap pr has some styling improvements, I'd like to merge this one if possible first and make the other two places with have these inline info links (indexes modal checkboxes and create db modal) use the new shared component there.

Syncing with design on what we want to do with these since as you said it's not one of LeafyGreen's intended patterns.  I'm alright with whatever approach we take here, I think the main part is having it shared in one place so whichever way we go we can make updates and styling changes in one place instead of many. And if we then remove it for another approach we easily know all the places where it's involved. Would it help do you think if I add a disclaimer to this `InlineInfoLink` class with some information on better usage (and to discourage use in a label).

Reply:
I think the fastest way to merge this would be to also remove the `label` attribute, for now, it may take a while before we would get feedback from DS about accepting elements for labels, maybe you can try to ask and see if is something it would be considered. If LG is not going in that direction then we should look for some other pattern to refactor labels

Reply:
Yup, updated this pr, once tests pass I'll merge and then finish up that bootstrap pr and ping you on it 👍